httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java

```diff
@@ -1033,22 +1033,22 @@ public URIBuilder normalizeSyntax() {
         if (this.pathSegments != null) {
             final List<String> inputSegments = this.pathSegments;
             if (!inputSegments.isEmpty()) {
-                final Stack<String> outputSegments = new Stack<>();
+                final LinkedList<String> outputSegments = new LinkedList<>();
```

Comment:
What is the benefit of this change?

Reply:
Performance. Stack extends Vector which is synchronized but LinkedList is not synchronized.

Reply:
Right, yes. This makes then sense.